src/app/modules/main/wallet_section/accounts/item.nim

```diff
@@ -125,3 +125,11 @@ QtObject:
 
   proc canSend*(self: Item): bool =
     return self.canSend
+
+  proc updateBalance*(self: Item, value: CurrencyAmount, assetsLoading: bool) =
+    self.assetsLoading = assetsLoading
+    self.currencyBalance = value
+    self.currencyBalanceChanged()
```

Comment:
I'm wondering why do we have this `self.currencyBalanceChanged`. Doesn't seem to be used in qml or nim

Could be a good chance to make the `WalletAccountItem` a simple object and remove this signal as probably there's no reason to inherit `QtObject`.

On the other hand, if there's a good reason to keep the signal, we should probably check if the value truly changed before sending the `changed` signal.

Reply:
I'm guessing it's because `currencyBalance` is a property here so it's a notifier signal

Reply:
yes, but I don't see it being used as is 🤔 . From what I see the data is provided by the model and the notifier should come in the form of `Model.dataChanged`.

Reply:
> I agree it could be refactored but I think it should be separate task

Sounds good! I've seen it being used, but doesn't look like this exact QObject. There's also `type` and other properties on that account instance. This `item.nim` seems to be exposed to qml only through the model and the model is in charge of the `dataChanged` signal. Probably all we need here is a simple nim object.